<launch>
  <arg name="model" default="$(find r2d2)/urdf/12-gripper.urdf.xacro"/>

  <include file="$(find r2d2)/launch/gazebo.launch">
    <arg name="model" value="$(arg model)" />
  </include>

  <rosparam command="load"
            file="$(find r2d2)/config/joints.yaml"
            ns="r2d2_joint_state_controller" />
  <rosparam command="load"
            file="$(find r2d2)/config/head.yaml"
            ns="r2d2_head_controller" />
  <rosparam command="load"
            file="$(find r2d2)/config/gripper.yaml"
            ns="r2d2_gripper_controller" />
  <rosparam command="load"
            file="$(find r2d2)/config/arm.yaml"
            ns="r2d2_arm_controller" />

  <node name="r2d2_controller_spawner" pkg="controller_manager" type="spawner"
    args="r2d2_joint_state_controller
          r2d2_head_controller
          r2d2_gripper_controller
          r2d2_arm_controller
          --shutdown-timeout 3"/>

  <!--<node name="gripper_controller_node" pkg="r2d2" type="gripper_controller"></node>-->
</launch>
